### PR TITLE
Attempt at example for logging consistency for discussion

### LINKED ISSUE #1137: Inconsistency in the logging messages
# Description
A lot of the logging messages are inconsistent. Let's take the file `loader.py` for example. On the line 254 there's that message that doesn't end with a dot:
```
_LOGGER.debug(
    _(
        "Couldn't find the command 'pip', "
        "trying again with command 'pip3'"
    )
)
```
Yet just a few lines ahead, on line 267, there's a message that does end with a dot:
```
_LOGGER.debug(
    _(
        "Couldn't find the command 'pip3', "
        "install of %s will be skipped."
    ),
    str(requirements_path),
)
```

There are also some inconsistencies with the parsers and connectors naming. In some logging messages they are capitalized properly while in the others they are not like in telegram's `__init__.py` file on lines 24 and 41:
```
_LOGGER.debug(_("Loaded telegram connector"))
```
```
_LOGGER.error(
    _(
        "Unable to login: Access token is missing. "
        "Telegram connector will be unavailable."
    )
)
```

I don't know for certain, but this might require some tweaking of the tests so they recognize the capital or lowercase letters properly.

I will tackle this issue if there will be no volunteers. :D

### PR DESCRIPTION
**Pull request by iguyking**
An quick run of connectors with a shot at rules around logging consistency.  

# Description

A quick run against connectors tree with a shot at something that might be useful for documenting consistency in logging.  

Fixes #1137

## Status
**UNDER DEVELOPMENT**

## Type of change

_Please delete options that are not relevant._

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update
- Documentation (fix or adds documentation)

# How Has This Been Tested?

Basic documentation updates effectively.  Should not be breaking type of changes unless there is some kind of width restriction concern for a given connector.  Most of my changes reduced the general length of the messages, with a couple just adding a character or two.  

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -181,6 +181,16 @@ Opsdroid is an open source chatbot framework. It is designed to be extendable, s
 
  If you don't know where to start, make sure to check the [tutorials](tutorials/introduction.md) and read the [documentation](http://opsdroid.readthedocs.io/en/latest/?badge=latest).  Remember that you can also ask for help in our [matrix channel](https://riot.im/app/#/room/#opsdroid-general:matrix.org)
 
+## Creating Consistent Logging Messages
+
+When creating log messages, we would like to see a consistency around how to structure and utilize them.
+
+* In general any copyright/legal name should be Capitalized (i.e. Slack, Telegram)
```
Comment (jacobtomlinson) [inline docs/contributing.md]: 👍

File: docs/contributing.md [hunk 2]
```diff
@@ -181,6 +181,16 @@ Opsdroid is an open source chatbot framework. It is designed to be extendable, s
...
+* All usage of the opsdroid capabilities should be Capitalized (Connector, Skill, Database and Parsers)
```
Comment (jacobtomlinson) [inline docs/contributing.md]: I had never considered this but I like it. That makes sense to me.

File: docs/contributing.md [hunk 3]
```diff
@@ -181,6 +181,16 @@ Opsdroid is an open source chatbot framework. It is designed to be extendable, s
...
+* Standalone sentances should not end with a period.  Use a period if there are more than one sentance in a line.
```
Comment (jacobtomlinson) [inline docs/contributing.md]: I'm not sure about this one. Why should folks avoid periods in standalone sentences. When viewed as a stream of logs this may seem inconsistent?

**Comment by jacobtomlinson** [comment]
Would also be good to get eyes from @opsdroid/maintainers on this.

File: docs/contributing.md (hunk 2 above)
Comment (iguyking) [reply docs/contributing.md]: This was based on how you recently did the Telegram update.  So figured this might be the direction you wanted to go in?

File: docs/contributing.md (hunk 3 above)
Comment (iguyking) [reply docs/contributing.md]: Mainly as each line was it's own message.   If its in a stream of logging, the newline would trigger the next thought/log message of event explanation. 

examples:
Line 1: 
Error:  running Skill dance
Line 2:
WARNING: Running Skill dance broke against name.  Check the name exists in the config file

vs 
Line 1:
Error: running Skill dance.  
Line 2:
WARNING: Running Skill dance broke against name.  Check the name exists in the config file.

I'm open to either, just kicking off the conversation.   The main thing is stating how we want it to be and stating it as such for the future.

Comment (jacobtomlinson) [reply docs/contributing.md]: Sure thanks! I think I prefer consistently having full stop.s

File: opsdroid/connector/rocketchat/__init__.py [hunk 4]
```diff
@@ -46,8 +46,8 @@ def __init__(self, config, opsdroid=None):
         except (KeyError, AttributeError):
             _LOGGER.error(
                 _(
-                    "Unable to login: Access token is missing. "
```
Comment (FabioRosado) [inline opsdroid/connector/rocketchat/__init__.py]: The space needs to be here since the  log will be logged as a single sentence/line. We simply wrote it like that to deal with linting issues when a line was longer than 79 characters. Now after we started using black, there are a lot of logging messages that got split up and should probably stay on the same line

File: docs/contributing.md (hunk 3 above)
Comment (FabioRosado) [reply docs/contributing.md]: We might be discussing very thin grain things here but I am with Jacob on this, I would rather have punctuation on the logs. Perhaps I'm wrong to assume that a lot of people don't care much about this since if you are looking at the logs it's because you need to debug something.

Although, keeping the logs consistent and have proper punctuation (even if it's only at the end of the line), might help in making things more obvious when reading the logs in the terminal.

After spending a few minutes debugging issues with my contributions, some times it can be hard to follow the logs in debug mode, but that full stop or other punctuation will automatically signal that this log line has ended.

File: opsdroid/connector/rocketchat/__init__.py (hunk 4 above)
Comment (iguyking) [reply opsdroid/connector/rocketchat/__init__.py]: What's the rule around having a space after the period?   Should all sentences ending with a period have a space?

File: docs/contributing.md (hunk 3 above)
Comment (iguyking) [reply docs/contributing.md]: I've updated my example set with the changes to be periods.    How does this look?   Ran into a unique scenario attempting to make everything a period ending sentence.  As such added a couple suggested lines to contributing doc page.

File: opsdroid/connector/rocketchat/__init__.py (hunk 4 above)
Comment (jacobtomlinson) [reply opsdroid/connector/rocketchat/__init__.py]: The period should be the last character. The problem here is that the string is made up of two strings on separate lines. This is a legacy thing that we don't need to do any more. If you put them together it will make more sense.

```python
"Unable to login: Access token is missing. Rocket.Chat connector will not be available."
```

File: opsdroid/connector/github/__init__.py [hunk 5]
```diff
@@ -35,7 +35,7 @@ def __init__(self, config, opsdroid=None):
         async with aiohttp.ClientSession() as session:
             response = await session.get(url)
             if response.status >= 300:
-                _LOGGER.error(_("Error connecting to github: %s"), response.text())
+                _LOGGER.error(_("Error connecting to GitHub: %s"), response.text())
```
Comment (jacobtomlinson) [inline opsdroid/connector/github/__init__.py]: Needs to end in a period.

File: opsdroid/connector/matrix/connector.py [hunk 6]
```diff
@@ -130,14 +130,14 @@ def filter_json(self):
                 # We can safely ignore timeout errors. The non-standard error
                 # codes are returned by Cloudflare.
                 if mre.code in [504, 522, 524]:
-                    _LOGGER.info(_("Matrix Sync Timeout (code: %d)"), mre.code)
+                    _LOGGER.info(_("Matrix sync timeout (code: %d)"), mre.code)
```
Comment (jacobtomlinson) [inline opsdroid/connector/matrix/connector.py]: Needs to end in a period.

File: opsdroid/connector/rocketchat/__init__.py (hunk 4 above)
Comment (iguyking) [reply opsdroid/connector/rocketchat/__init__.py]: Ran across a couple situations and not sure how we want to handle them.  

Example:  
_LOGGER.debug(payload)

There is no "text" of opsdroid in there.   Just the output of the error's payload.   

This was the idea behind the line of:

* When inserting the entire error output as the last part of the log message, do not end it with a period.

Just leave it or which way to address this?   Do we want to have _LOGGER.debug(payload, ".") 


Also.. 
  _LOGGER.debug(_("Unable to send image - " "Status Code %s."), resp.status)

Is this done to make translating easier?   Or should this be one string?

**Comment by iguyking** [comment]
Thoughts on the shift here @opsdroid/maintainers?

File: opsdroid/connector/rocketchat/__init__.py (hunk 4 above)
Comment (FabioRosado) [reply opsdroid/connector/rocketchat/__init__.py]: If we are passing the whole payload i'd say don't add the punctuation since this log will be mostly tech and we might wonder why the dot is there.

`_LOGGER.debug(_("Unable to send image - " "Status Code %s."), resp.status)`

This is another example of legacy logging, this two strings should be together like this:

`_LOGGER.debug(_("Unable to send image - Status Code %s."), resp.status)`

File: opsdroid/connector/github/__init__.py [hunk 7]
```diff
@@ -18,12 +18,12 @@ class ConnectorGitHub(Connector):
...
         """Create the connector."""
         super().__init__(config, opsdroid=opsdroid)
-        logging.debug("Loaded GitHub connector")
+        logging.debug("Loaded GitHub connector.")
         try:
             self.github_token = config["token"]
         except KeyError:
             _LOGGER.error(
-                _("Missing auth token!" "You must set 'token' in your config")
+                _("Missing auth token!" "You must set 'token' in your config.")
```
Comment (FabioRosado) [inline opsdroid/connector/github/__init__.py]: This should be a single string like such:
```suggestion
                _("Missing auth token! You must set 'token' in your config.")
```

Again this is legacy logging and seems like if we are touching it we should correct it 😄

File: opsdroid/connector/slack/__init__.py [hunk 8]
```diff
@@ -62,14 +70,14 @@ def __init__(self, config, opsdroid=None):
...
-            _LOGGER.info(_("Connected successfully"))
+            _LOGGER.debug(_("Connected as %s."), self.bot_name)
+            _LOGGER.debug(_("Using icon %s."), self.icon_emoji)
+            _LOGGER.debug(_("Default room is %s."), self.default_target)
+            _LOGGER.info(_("Connected successfully."))
         except slack.errors.SlackApiError as error:
             _LOGGER.error(
                 _(
-                    "Unable to connect to Slack due to %s - "
+                    "Unable to connect to Slack due to %s."
```
Comment (FabioRosado) [inline opsdroid/connector/slack/__init__.py]: This is another example that the two strings should be together like: "Unable to connect to Slack due to %s. The Slack Connector will not be available."

File: opsdroid/connector/telegram/__init__.py [hunk 9]
```diff
@@ -182,7 +181,7 @@ def build_url(self, method):
             ):
                 self.latest_update = result["update_id"] + 1
                 _LOGGER.debug(
-                    _("Emoji message parsing not supported " "- Ignoring message")
+                    _("Emoji message parsing not supported " "- Ignoring message.")
```
Comment (FabioRosado) [inline opsdroid/connector/telegram/__init__.py]: Logging should be together:

```suggestion
                    _("Emoji message parsing not supported - Ignoring message.")
```

File: opsdroid/connector/telegram/__init__.py [hunk 10]
```diff
@@ -212,13 +211,13 @@ def build_url(self, method):
                 _(
                     "Can't get updates because previous "
```
Comment (FabioRosado) [inline opsdroid/connector/telegram/__init__.py]: Sorry this one is even uglier than the other ones haha
this string should be:
`_("Can't get updates because previous webhook is still active. Will try to delete webhook.")`

File: opsdroid/connector/telegram/__init__.py [hunk 11]
```diff
@@ -294,9 +293,9 @@ def build_url(self, method):
 
         resp = await self.session.post(self.build_url("sendPhoto"), data=data)
         if resp.status == 200:
-            _LOGGER.debug(_("Sent %s image " "successfully"), file_event.name)
+            _LOGGER.debug(_("Sent %s image " "successfully."), file_event.name)
```
Comment (FabioRosado) [inline opsdroid/connector/telegram/__init__.py]:
```suggestion
            _LOGGER.debug(_("Sent %s image successfully."), file_event.name)
```

File: opsdroid/connector/telegram/__init__.py [hunk 12]
```diff
@@ -294,9 +293,9 @@ def build_url(self, method):
...
         else:
-            _LOGGER.debug(_("Unable to send image - " "Status Code %s"), resp.status)
+            _LOGGER.debug(_("Unable to send image - " "Status Code %s."), resp.status)
```
Comment (FabioRosado) [inline opsdroid/connector/telegram/__init__.py]:
```suggestion
            _LOGGER.debug(_("Unable to send image - Status Code %s."), resp.status)
```

File: opsdroid/core.py [hunk 13]
```diff
@@ -390,9 +390,9 @@ def _connector_names(self):  # noqa: D401
...
         try:
             if len(inspect.signature(skill).parameters.keys()) > 1:
-                await skill(self, config, event)
+                return await skill(self, config, event)
             else:
-                await skill(event)
+                return await skill(event)
         except Exception:
             _LOGGER.exception(
                 _("Exception when running skill '%s' "), str(config["name"])
```
Comment (FabioRosado) [inline opsdroid/core.py]:
```suggestion
                _("Exception when running skill '%s' ."), str(config["name"])
```

**Comment by jacobtomlinson** [comment]
@iguyking are you able to pick this up again?

**Comment by FabioRosado** [comment]
If you need help with updating the translations you can check the documentation on [localization](https://docs.opsdroid.dev/en/stable/contributing/#localization). I haven't tried to update the translations yet but I think you can just run the command:

```python setup.py update_catalog```

then compile them with the command: `python setup.py compile_catalog`

This should be enough, if you have any issue let me us know 😄

**Comment by jacobtomlinson** [comment]
Looks like we've still got a failing test here that is looking for a specific log message (that now has a period at the end and doesn't match).

Also black needs to be run.

@iguyking could you fix this couple of issues so we get a green tick and can merge?

File: opsdroid/connector/slack/__init__.py (hunk 8 above)
Comment (iguyking) [reply opsdroid/connector/slack/__init__.py]: Agreed.

**Comment by iguyking** [comment]
@opsdroid/maintainers  Fixed for the appveyor and travis-ci passing now.   Couple fixes around a test case and cleaning up more logging along with associated documentation.   Anything else?

File: opsdroid/loader.py [hunk 14]
```diff
@@ -303,6 +299,96 @@ def _load_intents(config):
...
+                return updated_configuration
+
+        except yaml.YAMLError as error:
+            _LOGGER.critical(error)
+            sys.exit(1)
+
+        except FileNotFoundError as error:
+            _LOGGER.critical(error)
+            sys.exit(1)
+
```
Comment (jacobtomlinson) [inline opsdroid/loader.py]: I think this shouldn't be here as it was removed in another PR. I'm guessing this has been reinstated due to some merge conflicts.

File: opsdroid/helper.py [hunk 15]
```diff
@@ -112,6 +112,48 @@ def update_pre_0_17_config_format(config):
...
+                "File %s copied from %s to %s. Run opsdroid -e to edit the main config file."
+            ),
+            file,
+            src,
+            dst,
+        )
+        if filecmp.cmp(original_file, copied_file):
+            os.remove(original_file)
+
+
```
Comment (jacobtomlinson) [inline opsdroid/helper.py]: I think this shouldn't be here as it was removed in another PR. I'm guessing this has been reinstated due to some merge conflicts.

**Comment by FabioRosado** [comment]
Yeah I can confirm that those bits where the ones I have either moved to configuration or removed it from the codebase

**Comment by jacobtomlinson** [comment]
Thanks @FabioRosado. I don't seem to be able to push onto this branch. @iguyking could you please delete the sections of code I highlighted and push again?

File: opsdroid/helper.py (hunk 15 above)
Comment (iguyking) [reply opsdroid/helper.py]: I'm pretty sure I messed that up.   When you push commits into these branches, how do I do merge conflicts properly?   It's obvious I did them wrong.

File: opsdroid/loader.py (hunk 14 above)
Comment (iguyking) [reply opsdroid/loader.py]: Don't be sorry.   I goofed up somehow.

**Comment by jacobtomlinson** [comment]
Thanks fo much for your patience here @iguyking! Glad to finally get this in.
